src/dotnet/commands/dotnet-build3/ForwardingApp.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class ForwardingApp
+    {
+        private static readonly string s_hostExe = "dotnet";
+
+        private string _forwardApplicationPath;
+        private string[] _argsToForward;
+        private string _depsFile;
+        private string _runtimeConfig;
+        private string _additionalProbingPath;
+        private Dictionary<string, string> _environment;
+
+        private string[] _allArgs;
+        
+        public ForwardingApp(
+            string forwardApplicationPath, 
+            string[] argsToForward, 
+            string depsFile = null, 
+            string runtimeConfig = null,
+            string additionalProbingPath = null,
+            Dictionary<string, string> environment = null)
+        {
+            _forwardApplicationPath = forwardApplicationPath;
+            _argsToForward = argsToForward;
+            _depsFile = depsFile;
+            _runtimeConfig = runtimeConfig;
+            _additionalProbingPath = additionalProbingPath;
+            _environment = environment;
+
+            _allArgs = new string[] { _forwardApplicationPath }.Concat(_argsToForward).ToArray();
+
+            if (_depsFile != null)
+            {
+                _allArgs = new string[] { "--depsfile", _depsFile }.Concat(_allArgs).ToArray();
+            }
+
+            if (_runtimeConfig != null)
+            {
+                _allArgs = new string[] { "--runtimeconfig", _runtimeConfig }.Concat(_allArgs).ToArray();
+            }
+
+            if (_additionalProbingPath != null)
+            {
+                _allArgs = new string[] { "--additionalprobingpath", _additionalProbingPath }.Concat(_allArgs).ToArray();
+            }
+
+            _allArgs = new string[] { "exec" }.Concat(_allArgs).ToArray();
+
+            Console.WriteLine(_forwardApplicationPath);
```

Comment:
So will it be removed?  I think putting it under `--verbose` is a good idea and keeping it in, that way we can debug in the future.